wayback/_client.py

```diff
@@ -525,7 +587,16 @@ def search(self, url, *, matchType=None, limit=None, offset=None,
                 break
 
             try:
-                data = CdxRecord(*text.split(' '), None, '', '')
+                data = CdxRecord(*text.split(' '), '', '')
+                if data.status_code == '-':
+                    # the status code given for a revisit record
+                    status_code = None
+                else:
+                    try:
+                        status_code = int(data.status_code)
+                    except TypeError as err:
+                        raise UnrecognizedStatusCode(status_code) from err
```

Comment:
The custom exception may be overzealous, but it will inject some potentially-useful specificity into the traceback. Commit dda82af19cd0e8fba4a1b6dd975afdc25f1b4434 makes the chaining explicit. The final exception will still be `UnexpectedResponseFormat`, which is accurate, but when applicable it will be clear that the direct cause was `UnrecognizedStatusCode`. I don't think we need custom exceptions for _every_ case. This just struck me as the riskiest bit of parsing that we do; I am hedging against the possibility that CDX might throw _other_ strings beyond `'-'` at us.

Reply:
I would also be totally fine with removing this extra exception in the name of code simplicity. Feel free to ask for that revision or push a commit of your own at will to expedite. :-)

Reply:
Ah, good point, I wasn’t thinking about the exception chaining aspect.

I don’t think I’m as convinced as you that status codes are a more special case, but I also don’t really feel strongly about it. Totally OK to merge as-is, I think! 👍

Reply:
Upon reflection, it's not worth a special exception type. Removed. Merging....